desktop/ui/src/main/java/org/datacleaner/widgets/tree/SchemaTree.java

```diff
@@ -645,9 +645,9 @@ public String getToolTipText(MouseEvent event) {
     public Datastore getDatastore() {
         return _datastore;
     }
-    
+
     @Override
-    public void onDescriptorsUpdated(DescriptorProvider descriptorProvider) {
+    public synchronized void onDescriptorsUpdated(DescriptorProvider descriptorProvider) {
```

Comment:
I doubt this is related to the issue you're trying to solve actually.

But _if_ you want to make a fix also for this here, then ```synchronized``` is not the right way. The potential issue you're facing is that a non-AWT thread is invoking this method (since it's a listener method - invoked by the descriptor provider monitoring thread). And all changes to AWT (and thereby Swing) must happen in the AWT dispatcher thread. So instead use WidgetUtils.invokeSwingAction(...).